src/terraform/command_helpers.go

```diff
@@ -29,11 +31,19 @@ func compileSecrets(secrets types.Secrets) []string {
 
 func compileEnvVars(deployConfig types.DeployConfig, secrets types.Secrets) ([]string, error) {
 	envVars := []string{}
+	dependencyEnvVars := map[string]string{}
+	for _, dependency := range deployConfig.AppConfig.Dependencies {
+		util.Merge(
+			dependencyEnvVars,
+			config.NewAppDependency(dependency, deployConfig.AppConfig, deployConfig.AppDir, deployConfig.HomeDir).GetDeploymentServiceEnvVariables(),
+		)
+	}
```

Comment:
Can we break these two for loops up into separate functions, or at least leave a comment? It's becoming difficult to parse what's going on here.